parsers.js

```diff
@@ -213,91 +143,101 @@ class Parsers {
+              board.trainServices.push(this.parseStandardService(service));
+            });
+          } catch (e) {
+            console.error(e);
+          }
           break;
-        case 'lt5:rsid':
-        case 'lt7:rsid':
-          train.rsid = element.val;
+        default:
+          console.log('unknown board element', element.name);
           break;
-        case 'lt5:origin':
-        case 'lt7:origin':
-          origin = element.childrenNamed('lt4:location');
-          train.origin = [];
-          origin.forEach((loc) => {
-            train.origin.push(this.parseLocation(loc));
+      }
+    });
+    return board;
+  }
+
+  static parseFormation(formation) {
+    const train = {};
+    formation.eachChild((element) => {
+      const name = element.name.split(':')[1];
+
+      switch (name) {
+        case 'coaches':
+          train.coaches = element.childrenNamed('lt7:coach').map((coach) => {
+            const coachData = {};
+            coachData.number = coach.attr.number;
+            coach.eachChild((el) => {
+              switch (el.name) {
+                case 'lt7:toilet':
+                  coachData.toilet = el.val;
+                  break;
+                case 'lt7:coachClass':
+                  coachData.class = el.val;
+                  break;
+                default:
+                  console.log('unknown coach element', el.name);
+                  break;
+              }
+            });
+            return coachData;
           });
           break;
-        case 'lt5:destination':
-        case 'lt7:destination':
-          destin = element.childrenNamed('lt4:location');
-          train.destination = [];
-          destin.forEach((loc) => {
-            train.destination.push(this.parseLocation(loc));
-          });
+        case 'avgLoading':
+          train.avgLoading = element.val;
           break;
         default:
+          console.log('unknown formation element', element.name);
           break;
       }
     });
     return train;
   }
 
+  static parseStandardService(service) {
+    const train = {};
+    service.eachChild((element) => {
+      const name = element.name.split(':')[1];
+      if (SERVICE_STRING_FIELDS.has(name)) {
+        train[name] = element.val;
+      } else if (SERVICE_BOOLEAN_FIELDS.has(name)) {
+        train[name] = element.val === 'true';
+      } else if (SERVICE_ARRAY_FIELDS.has(name)) {
+        train[name] = element.childrenNamed('lt4:location').map((loc) => this.parseLocation(loc));
+      } else if (name === 'formation') {
+        train[name] = this.parseFormation(element);
+      } else if (name === 'previousCallingPoints' || name === 'subsequentCallingPoints') {
+        const callingPoints = element.childNamed('lt7:callingPointList');
+        train[name] = this.parseCallingPointList(callingPoints);
+      } else {
+        console.log('unknown standard service element', element.name);
+      }
+    });
+    return train;
+  }
+
```

Comment:
Instead of switching through every element case, which was getting quite unwieldy after adding the new fields, I simplified it by combining all the ones where the type was the same